### PR TITLE
region properties. bbox_area returns image area instead of bbox area

### LINKED ISSUE #2634: `regioinprop.bbox_size` returns size of the original image
[This code](https://github.com/scikit-image/scikit-image/blob/master/skimage/measure/_regionprops.py#L125) returns the size of the original for all the regions.

### PR DESCRIPTION
**Pull request by sciunto**
## Description

Closes #2634

### CODE REVIEW

**Comment by sciunto** [comment]
I identified the problem with the test. In the test, we fall in a particular case where bbox area == image area. :(

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def bbox(self):
                      [self._slice[i].stop for i in range(self._ndim)])
 
     def bbox_area(self):
-        return self._label_image.size
+        return self.convex_image.size
```
Comment (soupault) [inline skimage/measure/_regionprops.py]: Should we implement it via call to `bbox` rather than `convex_image`?

Comment (sciunto) [inline skimage/measure/_regionprops.py]: I have no pro or cons. That one is just more concise.

Comment (jni) [inline skimage/measure/_regionprops.py]: Erm, `convex_image` is a computationally expensive feature to compute! Why is this change necessary at all?

Comment (jni) [inline skimage/measure/_regionprops.py]: Oh LOL never mind. But there are other better features for this, just a sec...

Comment (jni) [inline skimage/measure/_regionprops.py]: `self.image.size`?

Comment (soupault) [inline skimage/measure/_regionprops.py]: >Erm, convex_image is a computationally expensive feature to compute!

Exactly my thinking! I haven't checked in details, but I remeber that `regionprops` has a lot of lazy computations inside.

Comment (sciunto) [inline skimage/measure/_regionprops.py]: good lord... i haven't seen that it's computed once it's called...

Comment (sciunto) [inline skimage/measure/_regionprops.py]: I'll use bbox then...

File: skimage/measure/_regionprops.py [hunk 2]
```diff
@@ -123,7 +123,7 @@ def bbox(self):
                      [self._slice[i].stop for i in range(self._ndim)])
 
     def bbox_area(self):
-        return self._label_image.size
+        return (self.bbox[2] - self.bbox[0]) * (self.bbox[3] - self.bbox[1])
```
Comment (soupault) [inline skimage/measure/_regionprops.py]: Shouldn't this be generalized (3D/nD)?

Comment (jni) [inline skimage/measure/_regionprops.py]: Yes. Yes it should. =P

**Comment by daavve** [comment]
I believe the following is a generalized solution.

-Dave

```
diff --git a/skimage/measure/_regionprops.py b/skimage/measure/_regionprops.py
index 2cd5af67..0cd6d6fb 100644
--- a/skimage/measure/_regionprops.py
+++ b/skimage/measure/_regionprops.py
@@ -123,7 +123,7 @@ class _RegionProperties(object):
                      [self._slice[i].stop for i in range(self._ndim)])
 
     def bbox_area(self):
-        return self._label_image.size
+        return np.prod([self._slice[i].stop - self._slice[i].start for i in range(self._ndim)])
 
     def centroid(self):
         return tuple(self.coords.mean(axis=0))
```

File: skimage/measure/_regionprops.py (hunk 2 above)
Comment (daavve) [inline skimage/measure/_regionprops.py]: I believe the following is a generalized solution.

-Dave

```
diff --git a/skimage/measure/_regionprops.py b/skimage/measure/_regionprops.py
index 2cd5af67..0cd6d6fb 100644
--- a/skimage/measure/_regionprops.py
+++ b/skimage/measure/_regionprops.py
@@ -123,7 +123,7 @@ class _RegionProperties(object):
                      [self._slice[i].stop for i in range(self._ndim)])
 
     def bbox_area(self):
-        return self._label_image.size
+        return np.prod([self._slice[i].stop - self._slice[i].start for i in range(self._ndim)])
 
     def centroid(self):
         return tuple(self.coords.mean(axis=0))
```

**Comment by jni** [comment]
@sciunto thanks for this, but I've replaced this with a simpler nD solution in #2837. Closing this in the meantime. Thanks!
